File: core/java/src/main/java/hara/lang/protocol/IPushLast.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(namespace = "std.protocol.ipushlast", name = "IPushLast")
public interface IPushLast<E> {
  @HaraMethod(value = "push-last", arity = 2)
  IPushLast<E> pushLast(E e);
}

File: core/rust/src/lang/protocol/ipushlast.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(namespace = "std.protocol.ipushlast", name = "IPushLast")]
pub trait IPushLast<E> {
    type Output;

    #[hara_method(value = "push-last", arity = 2)]
    fn push_last(&self, value: E) -> Self::Output;
}
